docs/paper/dev/api/event-api/chat-event.md

```diff
@@ -0,0 +1,154 @@
+---
+slug: /dev/chat-events
+description: How to properly use the new AsyncChatEvent 
+---
+
+# Chat Events
+The chat event has evolved a couple of times over the years. 
+This guide will explain how to properly use the new `AsyncChatEvent` and its `Renderer`.
+
+## Understanding the Renderer
+
+Before we can start using the new chat event, we need to understand how the new renderer works.
+The renderer is Paper's way of allowing plugins to modify the chat message before it is sent to the player.
+This is done by using the `ChatRenderer` interface with its `render` method.
```

Comment:
I would maybe also mention somewhere that it is an improved replacement for the deprecated AsyncPlayerChatEvent#setFormat so authors who know of that can make a connection between the two.